libs/docs/platform/table/examples/preserved-state/platform-table-preserved-state-example.component.ts

```diff
@@ -0,0 +1,477 @@
+import { ChangeDetectionStrategy, Component, ViewChild, ViewEncapsulation } from '@angular/core';
+import { DatetimeAdapter, FdDate } from '@fundamental-ngx/core/datetime';
+import {
+    CollectionBooleanFilter,
+    CollectionCustomFilter,
+    CollectionDateFilter,
+    CollectionFilter,
+    CollectionGroup,
+    CollectionNumberFilter,
+    CollectionSelectFilter,
+    CollectionSort,
+    CollectionStringFilter,
+    FilterNumberStrategy,
+    FilterType,
+    SortDirection,
+    TableColumnsChangeEvent,
+    TableComponent,
+    TableDataProvider,
+    TableDataSource,
+    TableFilterChangeEvent,
+    TableFilterSelectOption,
+    TableGroupChangeEvent,
+    TablePageChangeEvent,
+    TableRowsRearrangeEvent,
+    TableRowToggleOpenStateEvent,
+    TableSortChangeEvent,
+    TableState
+} from '@fundamental-ngx/platform/table';
+import { map, Observable, of } from 'rxjs';
+
+@Component({
+    selector: 'fdp-platform-table-preserved-state-example',
+    templateUrl: './platform-table-preserved-state-example.component.html',
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class PlatformTablePreservedStateExampleComponent {
```

Comment:
I understand the purpose of this example is to illustrate the ability to save options regarding the state of the table, but I think we will see some confusion over the sort/group/filter not actually working here. Let's copy over sort/group/filter algorithms from other examples so it actually works here too